cheroot/test/test_ssl.py

```diff
@@ -757,3 +799,142 @@ def test_http_over_https_error(
         'The underlying error is {underlying_error!r}'.format(**locals())
     )
     assert expected_error_text in err_text
+
+
+@pytest.mark.parametrize(
+    ('adapter_type', 'encrypted_key', 'password_as_bytes'),
+    (
+        ('builtin', True, True),
+        ('builtin', True, False),
+        ('builtin', False, True),
+        ('builtin', False, False),
+        ('pyopenssl', True, True),
+        ('pyopenssl', True, False),
+        ('pyopenssl', False, True),
+        ('pyopenssl', False, False),
+    ),
+    ids=(
+        'builtin-encrypted-key-with-str-password',
+        'builtin-encrypted-key-with-bytes-password',
+        'builtin-non-encrypted-key-with-str-password',
+        'builtin-non-encrypted-key-with-bytes-password',
+        'pyopenssl-encrypted-key-with-str-password',
+        'pyopenssl-encrypted-key-with-bytes-password',
+        'pyopenssl-non-encrypted-key-with-str-password',
+        'pyopenssl-non-encrypted-key-with-bytes-password',
+    ),
+)
```

Comment:
This can be simplified to let pytest multiply the factors internally. Plus bytes/str IDs were swapped.
```suggestion
@pytest.mark.parametrize('adapter_type', ('builtin', 'pyopenssl'))
@pytest.mark.parametrize(
    'encrypted_key',
    (True, False),
    ids=('encrypted-key', 'unencrypted-key'),
)
@pytest.mark.parametrize(
    'password_as_bytes',
    (True, False),
    ids=('with-bytes-password', 'with-str-password'),
)
```

Reply:
Thank you, this was very helpful. Done in latest push.